packages/server/src/api/app/bsky/util/search.ts

```diff
@@ -0,0 +1,158 @@
+import { InvalidRequestError } from '@adxp/xrpc-server'
+import { sql } from 'kysely'
+import Database from '../../../../db'
+import { DbRef } from '../../../../db/util'
+
+export const getUserSearchQueryPg = (
+  db: Database,
+  opts: { term: string; limit: number; before?: string },
+) => {
+  const { ref } = db.db.dynamic
+  const { term, limit, before } = opts
+
+  // Performing matching by word using "strict word similarity" operator.
+  // The more characters the user gives us, the more we can ratchet down
+  // the distance threshold for matching.
+  const threshold = term.length < 3 ? 0.9 : 0.8
+  const cursor = before !== undefined ? unpackCursor(before) : undefined
+
+  // Matching user accounts based on username
```

Comment:
I think these queries are alright, but seeing some room for improvement in how they are executed.  Peeled that off into an issue we can poke at later: https://github.com/bluesky-social/adx/issues/239